pallets/rmrk-core/src/functions.rs

```diff
@@ -369,6 +373,10 @@ where
 
 		// TODO: Check NFT lock status
 
+		// Check NFT is transferable
```

Comment:
We could make this an internal public function returns `DispatchResult` so downstream can call a function like `is_transferable(&nft)`

Reply:
@HashWarlock Extracted function but passing actual transferable `bool` instead of nft reference. Wdyt?

Reply:
if that's the case, just change 377 to
```
		Self::check_is_transferable(&sending_nft)?;
```

and 697-701 to
```
	// Check NFT is transferable
	pub fn check_is_transferable(nft: &InstanceInfoOf<T>) -> DispatchResult {
		ensure!(nft.transferable, Error::<T>::NonTransferable);
		Ok(())
	}
```

Reply:
also would have to tweak rmrk-market lib.rs, 245-249:
```
			let nft = pallet_rmrk_core::Pallet::<T>::nfts(collection_id, nft_id)
				.ok_or(Error::<T>::TokenDoesNotExist)?;

			// Check NFT is transferable
			pallet_rmrk_core::Pallet::<T>::check_is_transferable(&nft)?;
```

Reply:
Agree, thanks for the tip.